### ISSUE TITLE
Default Analyers do not appear to run with 5.0 RC 2 SDK

### ISSUE THREAD

**Issue by mtaylorfsmb** [dotnet/roslyn-analyzers#4351]
### Analyzer

All of them.

### Analyzer source

**SDK**:  5.0.100-rc.2.20479.15 

**Version**:  5.0.100-rc.2.20479.15 

**Version**: 5.0.0-rc.2.20475.17

### Describe the bug

I might be missing something here but following the documentation on having the built in NET Analyzers run on my code isn't working and I have a NET 5.0 ASP.NET app that I'm testing with. It is similar to the issue that was reported [here](https://github.com/dotnet/roslyn-analyzers/issues/4165) that is now closed but my project is .NET 5.

I have code in a project that should generate CA warnings such as CA1062 and CA2208. If I have the FxCopAnalyzers NuGet package installed then the warnings are generated. If I remove that package then they aren't generated anymore. 

My understanding of the documentation is that if I'm targeting .NET 5.0 Preview 8 or higher (I have RC 2) then the SDK that is being referenced will automatically include the NETAnalyzers which include these rules. But with the FxCop NuGet package removed I don't see any CA warnings/suggestions/errors.

Note that the IDE shows the quick action for the CA issues when the NuGet package is included but with it gone they go away as  well.

### Steps To Reproduce

Sample `directory.build.props` file applied at solution level.

```xml
<Project>
   <PropertyGroup>
      <EnableNETAnalyzers>true</EnableNETAnalyzers>
      <AnalysisLevel>latest</AnalysisLevel>
   </PropertyGroup>
</Project>
```

Sample project file for ASP.NET 5 project.

```xml
<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
     <TargetFramework>net5.0</TargetFramework>     
  </PropertyGroup>
</Project>
```

Sample file that should have CA1062 issue.

```cs
using System;
using Microsoft.AspNetCore.Mvc;

namespace Test.Controllers
{
    /// <summary>Base class for controllers.</summary>
    public abstract class TestControllerBase : ControllerBase
    {        
        protected CreatedAtActionResult CreatedAtActionFixed ( string actionName, string controllerName, object routeValues, object model = null )
        {
            //Expecting CA1062 here...
            if (actionName.EndsWith("Async", StringComparison.OrdinalIgnoreCase))
                actionName = actionName.Left(actionName.Length - 5);

            var hasController = !String.IsNullOrEmpty(controllerName);
            if (hasController && controllerName.EndsWith("Controller", StringComparison.OrdinalIgnoreCase))
                controllerName = controllerName.Left(controllerName.Length - 10);

            return hasController ? CreatedAtAction(actionName, controllerName, routeValues, model) :
                        CreatedAtAction(actionName, routeValues, model);
        }
    }
}
```

### Expected behavior

Should see CA issues when targeting .NET 5 using RC2 without the NuGet analyzer packages directly added to the project.

### Actual behavior

Without the NuGet analyzer packages no CA issues are reported in the IDE or at build time.

**Comment by mavasani** [comment]
This is by design.

Please read about default enabled rules in the SDK here: https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/overview#enabled-rules

Please read about enabling additional rules here: https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/overview#enable-additional-rules

**Comment by mtaylorfsmb** [comment]
I'm sorry but this is a mess and the documentation is insufficient here. In the paragraph you linked to it says the default enabled rules and in the very next block it indicates that RC2 ships with all the rules. Nobody outside the people who wrote these analyzers would understand this. This needs to be far better documented.

At a minimum, given that most people either don't care or are currently using the FxCop analyzers, there needs to be a section that discusses how to migrate from the "old" approach to the "new" approach with a step by step guide as to what needs to be done. As part of that guide it needs to clarify that most of the existing analyzer rules (that are on in the NuGet package) are not on in the SDK. Therefore there is going to be a manual process for enabling them all. 

The introductory paragraph for this page even says that starting in .NET 5 these are included in the SDK and that previously you used the NuGet package. While technically correct this is misleading because they aren't on by default so people will read this as not needing the analyzers anymore and removing them not realizing they just stopped using the bulk of them. A clear step-by-step process needs to be provided here. 

The fact that projects/solutions will now have to "customize" the bulk of the analyzer rules they are used to having before is going to result in, I suspect, a lot of people sticking with the NuGet package. At this point there is no benefit in having the SDK support the analyzer rules in my opinion.

**Comment by mavasani** [comment]
> I'm sorry but this is a mess and the documentation is insufficient here.

Please file a doc issue.

> In the paragraph you linked to it says the default enabled rules and in the very next block it indicates that RC2 ships with all the rules. Nobody outside the people who wrote these analyzers would understand this. This needs to be far better documented.

Shipping a rule does not mean it is enabled by default. I think the documentation is pretty clear, but if you disagree please raise a doc feedback request on that page.

> At a minimum, given that most people either don't care or are currently using the FxCop analyzers

That is absolutely not true - as per telemetry, less then 2% of VS .NET users are using or aware about FxCop analyzers.

> there needs to be a section that discusses how to migrate from the "old" approach to the "new" approach with a step by step guide as to what needs to be done

Remove package reference to FxCop analyzers NuGet package and switch to Aggressive/opt-out custom analysis mode: https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/overview#custom-analysis-mode

> The fact that projects/solutions will now have to "customize" the bulk of the analyzer rules they are used to having before is going to result in, I suspect, a lot of people sticking with the NuGet package. At this point there is no benefit in having the SDK support the analyzer rules in my opinion.

You are free to stay with the NuGet package if you wish. Analyzers included in the SDK are **primarily for .NET users who are not even aware of such an analyzer NuGet package**, and we don't want a deluge of new analyzer warnings for users just moving to the .NET 5.0 SDK. Most critical analyzers are enabled by default as warnings to ensure a balance of noise vs warnings.
